central/policycategory/service/service_impl.go

```diff
@@ -0,0 +1,132 @@
+package service
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/policycategory/datastore"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.Policy)): {
+			"/v1.PolicyCategoryService/GetPolicyCategory",
+			"/v1.PolicyCategoryService/GetPolicyCategories",
+		},
+		user.With(permissions.Modify(resources.Policy)): {
+			"/v1.PolicyCategoryService/PostPolicyCategory",
+			"/v1.PolicyCategoryService/RenamePolicyCategory",
+			"/v1.PolicyCategoryService/DeletePolicyCategory",
+		},
+	})
+)
+
+var (
+	policySyncReadCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Policy)))
+)
+
+// serviceImpl provides APIs for alerts.
+type serviceImpl struct {
+	policyCategoriesDatastore datastore.DataStore
+}
+
+func (s *serviceImpl) GetPolicyCategory(ctx context.Context, id *v1.ResourceByID) (*v1.PolicyCategory, error) {
+	return s.getPolicyCategory(ctx, id.GetId())
+}
+
+func (s *serviceImpl) GetPolicyCategories(ctx context.Context, query *v1.RawQuery) (*v1.GetPolicyCategoriesResponse, error) {
+	resp := new(v1.GetPolicyCategoriesResponse)
+
+	if query.GetQuery() == "" {
+		categories, err := s.policyCategoriesDatastore.GetAllPolicyCategories(ctx)
+		if err != nil {
+			return nil, err
+		}
+		resp.Categories = s.convertCategoriesToV1Categories(categories)
+
+	} else {
+		parsedQuery, err := search.ParseQuery(query.GetQuery())
```

Comment:
Does the searcher not work if the query is empty? If it does, this could be simplified.

Reply:
Ack, yes! Done.